src/coreclr/vm/jitinterface.cpp

```diff
@@ -8741,44 +8741,15 @@ CORINFO_CLASS_HANDLE CEEInfo::getDefaultComparerClassHelper(CORINFO_CLASS_HANDLE
     if (Nullable::IsNullableType(elemTypeHnd))
     {
         Instantiation nullableInst = elemTypeHnd.AsMethodTable()->GetInstantiation();
-        TypeHandle iequatable = TypeHandle(CoreLibBinder::GetClass(CLASS__IEQUATABLEGENERIC)).Instantiate(nullableInst);
-        if (nullableInst[0].CanCastTo(iequatable))
-        {
-            TypeHandle resultTh = ((TypeHandle)CoreLibBinder::GetClass(CLASS__NULLABLE_COMPARER)).Instantiate(nullableInst);
-            return CORINFO_CLASS_HANDLE(resultTh.GetMethodTable());
-        }
+        TypeHandle resultTh = ((TypeHandle)CoreLibBinder::GetClass(CLASS__NULLABLE_COMPARER)).Instantiate(nullableInst);
+        return CORINFO_CLASS_HANDLE(resultTh.GetMethodTable());
     }
 
     // We need to special case the Enum comparers based on their underlying type to avoid boxing
     if (elemTypeHnd.IsEnum())
     {
-        MethodTable* targetClass = NULL;
-        CorElementType normType = elemTypeHnd.GetVerifierCorElementType();
-
-        switch(normType)
-        {
-            case ELEMENT_TYPE_I1:
-            case ELEMENT_TYPE_I2:
-            case ELEMENT_TYPE_U1:
-            case ELEMENT_TYPE_U2:
-            case ELEMENT_TYPE_I4:
-            case ELEMENT_TYPE_U4:
-            case ELEMENT_TYPE_I8:
-            case ELEMENT_TYPE_U8:
-            {
-                targetClass = CoreLibBinder::GetClass(CLASS__ENUM_COMPARER);
-                break;
-            }
-
-            default:
-                break;
-        }
-
-        if (targetClass != NULL)
-        {
-            TypeHandle resultTh = ((TypeHandle)targetClass->GetCanonicalMethodTable()).Instantiate(inst);
-            return CORINFO_CLASS_HANDLE(resultTh.GetMethodTable());
-        }
+        TypeHandle resultTh = ((TypeHandle)CoreLibBinder::GetClass(CLASS__ENUM_COMPARER)).Instantiate(inst);
+        return CORINFO_CLASS_HANDLE(resultTh.GetMethodTable());
```

Comment:
Also delete the matching special handling for `METHOD__RUNTIME_HELPERS__ENUM_COMPARE_TO`  earlier in this file?

Reply:
Is the intrinsic even needed there anymore? Since now there's the `Enum` constraint and those have managed IMPLs:
https://github.com/dotnet/runtime/blob/bfa9b97712a3c62fba189c651a25f6d91ff3499f/src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs#L212-L226

Reply:
The managed implementation is slow boxing implementation. The intrinics impl is for perf.

Reply:
> Also delete the matching special handling for `METHOD__RUNTIME_HELPERS__ENUM_COMPARE_TO` earlier in this file?

The CompareTo special handling seems tricky to remove due to this relying on the CorElementType layout:
https://github.com/dotnet/runtime/blob/bfa9b97712a3c62fba189c651a25f6d91ff3499f/src/coreclr/vm/jitinterface.cpp#L7282-L7295

I think we should leave it there for now.

Reply:
I meant to replace this: https://github.com/dotnet/runtime/blob/bfa9b97712a3c62fba189c651a25f6d91ff3499f/src/coreclr/vm/jitinterface.cpp#L7273-L7280 with just `if (inst[0].IsEnum())`

Reply:
> with just `if (inst[0].IsEnum())`

That wouldn't work since that code then indexes the table I've linked with `et`, so it'll then access it out of bounds.

Reply:
I'll fix it in the next PR that'll make the `RuntimeHelpers` intrinsics JIT ones, is that okay? @jkotas